rtl/eclrtl/eclhelper_dyn.cpp

```diff
@@ -136,10 +136,8 @@ class FilterSet
         if (fieldNum == (unsigned) -1)
             throw MakeStringException(0, "Invalid filter string: field '%s' not recognized", fieldName.str());
         unsigned numOffsets = inrec.getNumVarFields() + 1;
-        size_t * variableOffsets = (size_t *)alloca(numOffsets * sizeof(size_t));
-        RtlRow offsetCalculator(inrec, nullptr, numOffsets, variableOffsets);
-        unsigned fieldOffset = offsetCalculator.getOffset(fieldNum);
-        unsigned fieldSize = offsetCalculator.getSize(fieldNum);
+        unsigned fieldOffset = inrec.getFixedOffset(fieldNum);
```

Comment:
This change not directly related, but found in testing that the filter feature in dumpkey had been broken by the changes to lazy field offset calculation. Code was a bit dodgy before and is equally dodgy now.